packages/flutter_test/lib/src/recording_canvas.dart

```diff
@@ -170,7 +173,18 @@ class TestRecordingPaintingContext extends ClipContext implements PaintingContex
     canvas.saveLayer(null, Paint()); // TODO(ianh): Expose the alpha somewhere.
     painter(this, offset);
     canvas.restore();
-    return OpacityLayer();
+    final OpacityLayer layer = OpacityLayer();
+    _createdLayers.add(layer);
```

Comment:
As I said on https://github.com/flutter/flutter/pull/134701#discussion_r1326534750, this is likely not the right fix. If I remember correctly, the contract of PaintingContext (which defines this method) is that the caller assumes ownership of the Layer and is in charge of disposing it. If this change is in fact "fixing" any leaks, it is only doing so for the test. In production (where a regular PaintContext and not the testing version is used) this leak is still present!